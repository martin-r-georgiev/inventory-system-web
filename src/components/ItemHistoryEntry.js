import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleDoubleUp } from '@fortawesome/free-solid-svg-icons'
import { faAngleDoubleDown } from '@fortawesome/free-solid-svg-icons'

const ItemHistoryEntry = ({timestamp, quantity, trend}) => {
    let formattedTime = new Date(timestamp).toUTCString();

    let trendIcon = "";
    if (trend)
    {
        if(trend === "up") trendIcon = (<FontAwesomeIcon icon={faAngleDoubleUp} color="green" style={{fontSize: "20px"}}/>);
        else { trendIcon = (<FontAwesomeIcon icon={faAngleDoubleDown} color="red" style={{fontSize: "20px"}}/>); }
    }

    return(
    <div className="shadow-sm p-3 mb-5 bg-white rounded entry-wrapper">
        <div className="item-history-entry">
            <div data-testid="item-entry-body" className="row d-flex align-items-center justify-content-between">
                <div className="col-10 col-sm d-flex align-items-center">
                    {formattedTime}
                </div>
                <div className={`col-1 col-sm-1 p-2 text-white d-flex justify-content-center ${quantity > 0 ? "bg-success" : "bg-danger"} entry-quantity`}>
                    {quantity}
                </div>
                <div className="col-1 col-sm-1 d-flex justify-content-center">
                    {trendIcon}
                </div>
            </div>  
        </div>
    </div>
    
    );
};

export default ItemHistoryEntry;