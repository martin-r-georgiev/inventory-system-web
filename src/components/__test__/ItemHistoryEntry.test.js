import React from 'react';
import ReactDOM from 'react-dom';
import { itTSAnyKeyword } from '@babel/types';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';

import ItemHistoryEntry from '../ItemHistoryEntry';

afterEach(cleanup);

it("Renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<ItemHistoryEntry/>, div);
});

it("Renders ItemHistoryEntry component correctly", () => {
    let timeNow = new Date(2000, 1, 1);
    let formattedTime = timeNow.toUTCString();

    const {getByTestId} = render(<ItemHistoryEntry timestamp={timeNow} quantity="40" trend="up"/>);

    expect(getByTestId('item-entry-body')).toHaveTextContent(formattedTime.toString());
    expect(getByTestId('item-entry-body')).toHaveTextContent("40");
});

it("Matches snapshot", () => {
    const tree = renderer.create(<ItemHistoryEntry timestamp={Date(2000, 1, 1)} quantity="40" trend="up"/>).toJSON();

    expect(tree).toMatchSnapshot();
});