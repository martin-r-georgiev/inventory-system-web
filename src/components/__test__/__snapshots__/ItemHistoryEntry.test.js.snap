// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="shadow-sm p-3 mb-5 bg-white rounded entry-wrapper"
>
  <div
    className="item-history-entry"
  >
    <div
      className="row d-flex align-items-center justify-content-between"
      data-testid="item-entry-body"
    >
      <div
        className="col-10 col-sm d-flex align-items-center"
      >
        Fri, 22 Jan 2021 01:37:28 GMT
      </div>
      <div
        className="col-1 col-sm-1 p-2 text-white d-flex justify-content-center bg-success entry-quantity"
      >
        40
      </div>
      <div
        className="col-1 col-sm-1 d-flex justify-content-center"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-angle-double-up fa-w-10 "
          color="green"
          data-icon="angle-double-up"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={
            Object {
              "fontSize": "20px",
            }
          }
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
